service/storage_service.cc

```diff
@@ -1992,8 +1992,12 @@ class topology_coordinator {
                 // node into `exclude_nodes` parameter in `exec_global_command`, but CDC generations are never
                 // introduced during replace/remove.
                 {
+                    raft_topology_cmd cmd{raft_topology_cmd::command::barrier};
+                    for (const auto& [id, _]: _topo_sm._topology.transition_nodes) {
+                        cmd.wait_for_ip.push_back(id);
+                    }
```

Comment:
Why do you need to pass it explicitly. The receiving node can look at transition_nodes by itself.

Reply:
I wanted to have the IP mapping inplace before `read_barrier`/`topology_state_load`, so that `topology_state_load` could add the proper mapping into erm. Although on the other hand now I'm moving erm to `raft_address_map` and this requirement is no longer relevant.

Reply:
rewrote without explicit parameter, use `_topology_state_machine._topology.new_nodes`